server/application/application.proto

```diff
@@ -22,9 +22,7 @@ message ApplicationResponse {}
 
 message DeleteApplicationRequest {
 	string name = 1;
-	string namespace = 2;
-	string server = 3;
-	bool force = 4;
+	bool force = 2;
```

Comment:
The term force is unclear what will happen in the delete request. This flag really indicates whether or not to perform cascaded deletion or not.

Can we rename this to `cascade` which can either be nil, true or false?

I think the semantics should be:
* If cascade is false, remove the finalizer (if present). delete the object directly from kubernetes. this should not involve the controller.
* If cascade is true, set the finalizer (if not already set). delete the object directly from kubernetes. deletion will be blocked until the finalizer is removed by the controller
* If cascade is nil. don't touch finalizers one way or the other. delete the object directly from kubernetes. deletion may or may not involve controller depending on if finalizer was already set.

Reply:
I think to support nullable fields you may need (gogoproto.nullable) = true.